patricia/patricia.go

```diff
@@ -19,6 +19,29 @@ type (
 	VisitorFunc func(prefix Prefix, item Item) error
 )
 
+const (
+	DefaultMaxPrefixPerNode         = 10
+	DefaultMaxChildrenPerSparseNode = 8
+)
+
+type Option struct {
+	// Max prefix length that is kept in a single trie node.
+	MaxPrefixPerNode int
+	// Max children to keep in a node in the sparse mode.
+	MaxChildrenPerSparseNode int
+}
+
+func NewOption(maxPrefixPerNode int, maxChildrenPerSparseNode int) *Option {
```

Comment:
Not sure I want to be sorting the list every time there is something added... In any case, the behavior is not really defined I would say.